usersession/userd/launcher_internal_test.go

```diff
@@ -0,0 +1,119 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2019 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package userd
+
+import (
+	"github.com/snapcore/snapd/strutil"
+	"reflect"
+	"strings"
+	"testing"
+)
```

Comment:
We use `gopkg.in/check.v1` for testing. Please take a look at `usersession/autostart/autostart_test.go` for an example.

The test suite would look like this:

```
import (
	. "gopkg.in/check.v1"

	"github.com/snapcore/snapd/dirs"

)

func Test(t *testing.T) { TestingT(t) }

type launcherSuite struct {
	dir string
}

var _ = Suite(&autostartSuite{})

func (s *launcherSuite) SetUpTest(c *C) {
	s.dir = c.MkDir()
	dirs.SetRootDir(s.dir)
	
	// now your dirs.SnapdDesktopFilesDir is mocked in tests
}

func (s *launcherSuite) TearDownTest(c *C) {
	s.dir = c.MkDir()
	dirs.SetRootDir("/")
}

func (s *launcherSuite) TestFoo(c *C) {
	// mock *.desktop file with some content
	err := ioutil.WriteFile(filepath.Join(dirs.SnapDesktopFilesDir, "foo.desktop", []byte("some content"), 0644)
	c.Assert(err, IsNiil)
	...
}

```